handlers/api.go

```diff
@@ -372,37 +372,54 @@ func ApiEpochSlots(w http.ResponseWriter, r *http.Request) {
 }
 
 // ApiSlots godoc
-// @Summary Get a slot by its slot number or root hash
+// @Summary Get a slot by its slot number or root hash. Alternatively get the latest slot or the slot containing the head block.
 // @Tags Slot
-// @Description Returns a slot by its slot number or root hash or the latest slot with string latest
+// @Description Returns a slot by its slot number or root hash, the latest slot with string latest or the slot containing the head block with string head
 // @Produce  json
-// @Param  slotOrHash path string true "Slot or root hash or the string latest"
+// @Param  slotOrHash path string true "Slot or root hash or the string latest or head"
 // @Success 200 {object} types.ApiResponse{data=types.APISlotResponse}
 // @Failure 400 {object} types.ApiResponse
 // @Router /api/v1/slot/{slotOrHash} [get]
 func ApiSlots(w http.ResponseWriter, r *http.Request) {
 	w.Header().Set("Content-Type", "application/json")
 
 	vars := mux.Vars(r)
-
 	slotOrHash := strings.Replace(vars["slotOrHash"], "0x", "", -1)
+
 	blockSlot := int64(-1)
-	blockRootHash, err := hex.DecodeString(slotOrHash)
-	if slotOrHash != "latest" && (err != nil || len(slotOrHash) != 64) {
-		blockRootHash = []byte{}
-		blockSlot, err = strconv.ParseInt(vars["slotOrHash"], 10, 64)
-		if err != nil {
-			sendErrorResponse(w, r.URL.String(), "could not parse slot number")
-			return
-		}
-	}
+	blockRootHash := []byte{}
 
 	if slotOrHash == "latest" {
+		// simply check the latest slot (might be empty which causes an error)
 		blockSlot = int64(services.LatestSlot())
+	} else if slotOrHash == "head" {
+		// retrieve the slot containing the head block of the chain (cannot return an empty slot unless there is no block in the chain yet)
+		err := db.ReaderDb.Get(&blockRootHash, `
```

Comment:
this should not hit the database, instead create a new property in the services module that gets updated via the FDU

Reply:
Added a new service accordingly